nxc/protocols/smb.py

```diff
@@ -356,6 +356,7 @@ def laps_search(self, username, password, ntlm_hash, domain):
             hash_ntlm = hashlib.new("md4", msMCSAdmPwd.encode("utf-16le")).digest()
             self.hash = binascii.hexlify(hash_ntlm).decode()
 
+        self.args.local_auth = True
```

Comment:
We should probably make this an object attribute instead of editing the args object directly and/or make an attribute like "bh_enabled" for the specific action of adding to BH, since local_auth might be used elsewhere, and we don't want to edit it like this if other checks rely on this to be correct.

Reply:
hmm yeah would be an idea. Although we would have to rework when credentials get added to bloodhound because there is just the check for `--local.auth`

Reply:
Yeah we probably want to separate those, since there might be sometime that a host isnt local auth but we don't want to add it to BH. Not sure exactly _what_ that could be, but it's not crazy to consider.

Reply:
Hmm atleast about the `--local-auth` part i think we should set the arg to that. If you want to sign in with `--laps` it is more or less getting the local admin accounts pw from ldap and then signing in **locally** with that account, most often the administrator. So logically atleast imo adding `--laps` also implicates a local authentication in the end with the local user and his password, whatever comes with it. So for me `--laps` would implicate `--local-auth` anyway, or is there a logic flaw i am missing?

For the bloodhound part, maybe there is some edge case where we wouldnt want to add a user to bloodhound besides `local-auth`, but no idea what that would be. Or more like i can't think of a situation where you would want to add local account creds to bloodhound. Or is there?

Reply:
Neff and I discussed offline, I had a misunderstanding about what this does. Looks good.